util/fipstools/inject_hash/inject_hash.go

```diff
@@ -170,51 +122,252 @@ func do(outPath, oInput string, arInput string, useSHA256 bool) error {
+		return nil, nil, errors.New("rodata start marker inconsistent with rodata section presence")
 	}
 
 	if (rodataStart != nil) != (rodataEnd != nil) {
-		return errors.New("rodata marker presence inconsistent")
+		return nil, nil, errors.New("rodata marker presence inconsistent")
 	}
 
 	if max := textSection.Size; *textStart > max || *textStart > *textEnd || *textEnd > max {
-		return fmt.Errorf("invalid module .text boundaries: start: %x, end: %x, max: %x", *textStart, *textEnd, max)
+		return nil, nil, fmt.Errorf("invalid module .text boundaries: start: %x, end: %x, max: %x", *textStart, *textEnd, max)
 	}
 
 	if rodataSection != nil {
 		if max := rodataSection.Size; *rodataStart > max || *rodataStart > *rodataEnd || *rodataEnd > max {
-			return fmt.Errorf("invalid module .rodata boundaries: start: %x, end: %x, max: %x", *rodataStart, *rodataEnd, max)
+			return nil, nil, fmt.Errorf("invalid module .rodata boundaries: start: %x, end: %x, max: %x", *rodataStart, *rodataEnd, max)
 		}
 	}
 
 	// Extract the module from the .text section and hash it.
 
 	text := textSection.Open()
 	if _, err := text.Seek(int64(*textStart), 0); err != nil {
-		return errors.New("failed to seek to module start in .text: " + err.Error())
+		return nil, nil, errors.New("failed to seek to module start in .text: " + err.Error())
 	}
 	moduleText := make([]byte, *textEnd-*textStart)
 	if _, err := io.ReadFull(text, moduleText); err != nil {
-		return errors.New("failed to read .text: " + err.Error())
+		return nil, nil, errors.New("failed to read .text: " + err.Error())
 	}
 
 	// Maybe extract the module's read-only data too
 	var moduleROData []byte
 	if rodataSection != nil {
 		rodata := rodataSection.Open()
 		if _, err := rodata.Seek(int64(*rodataStart), 0); err != nil {
-			return errors.New("failed to seek to module start in .rodata: " + err.Error())
+			return nil, nil, errors.New("failed to seek to module start in .rodata: " + err.Error())
 		}
 		moduleROData = make([]byte, *rodataEnd-*rodataStart)
 		if _, err := io.ReadFull(rodata, moduleROData); err != nil {
-			return errors.New("failed to read .rodata: " + err.Error())
+			return nil, nil, errors.New("failed to read .rodata: " + err.Error())
 		}
 	}
 
+	return moduleText, moduleROData, nil
+}
+
+
+func doMacOS(objectBytes []byte) ([]byte, []byte, error) {
+
+	object, err := macho.NewFile(bytes.NewReader(objectBytes))
+	if err != nil {
+		return nil, nil, errors.New("failed to parse object: " + err.Error())
+	}
+
+	// Find the __text and, optionally, __const sections.
+	var textSection, rodataSection *macho.Section
+	var textSectionIndex, rodataSectionIndex int
+	for i, section := range object.Sections {
+
+    if section.Seg == "__TEXT" && section.Name == "__text" {
+			textSection = section
+			textSectionIndex = i + 1
+    }
+    if section.Seg == "__TEXT" && section.Name == "__const" {
+			rodataSection = section
+			rodataSectionIndex = i + 1
+    }
```

Comment:
indentation. Is there at most one of each? If yes, may be add that to a comment?

Reply:
done